R/inferCNV.R

```diff
@@ -141,29 +151,44 @@ CreateInfercnvObject <- function(raw_counts_matrix, gene_order_file, annotations
         stop(error_message)
     }
 
-    # restrict expression data to the annotated cells.
+    if (! is.null(max_cells_per_group)) {
+        ## trim down where needed.
+        grps = split(input_classifications, input_classifications[,1])
+        newdf = NULL
+        for (grp in names(grps)) {
+            df = grps[[grp]]
+            if (dim(df)[1] > max_cells_per_group) {
+                flog.info(sprintf("-reducing number of cells for grp %s from %g to %g",
+                                  grp, dim(df)[1], max_cells_per_group))
+                grps[[grp]] = df[sample(1:dim(df)[1], max_cells_per_group),,drop=F]
+            }
+        }
+        input_classifications = data.frame(Reduce(rbind, grps))
+    }
+        
+    ## restrict expression data to the annotated cells.
     raw.data <- raw.data[,colnames(raw.data) %in% rownames(input_classifications)]
         
-    # reorder cell classifications according to expression matrix column names
+    ## reorder cell classifications according to expression matrix column names
     input_classifications <- input_classifications[order(match(row.names(input_classifications), colnames(raw.data))), , drop=FALSE]
 
     
-    # get indices for reference cells
+    ## get indices for reference cells
     ref_group_cell_indices = list()
     for (name_group in ref_group_names) {
         cell_indices = which(input_classifications[,1] == name_group)
+        cell_names =  rownames(input_classifications)[cell_indices]
```

Comment:
Does not appear to be used anywhere in this function, can it be removed? Same applies for line 219.